src/detext/train/metrics.py

```diff
@@ -54,6 +54,34 @@ def update_state(self, labels, scores, sample_weight=None):
         return super(AUCMetric, self).update_state(labels, prob, sample_weight)
 
 
+class PrecisionMetric(tf.keras.metrics.Precision):
+    """ Metric for computing Precision given DeText format input """
+
+    def update_state(self, labels, scores, sample_weight=None):
+        """ Accumulates metric statistics
+
+        :param scores: Tensor Predicted scores as logits. Shape=[batch_size, max_group_size(1)] for ranking. For classification, shape=[batch_size]
+        :param labels: Tensor Labels. Shape=[batch_size, max_group_size(1)] for ranking. For binary classification, shape=[batch_size]
+        :param sample_weight: Sample weight. Check the inherited class method for more detail
+        """
+        probabilities = tf.round(tf.nn.sigmoid(scores))  # Convert to probabilities in range [0, 1]
```

Comment:
After rounding the probabilities will become predictions, please change the variable name.